pennylane_lightning/lightning_tensor/backends/quimb/_mps.py

```diff
@@ -79,22 +214,215 @@ def __init__(self, num_wires, interf_opts, dtype=np.complex128):
+
+        Args:
+            circuit (QuantumScript): The single circuit to simulate.
+
+        Returns:
+            Tuple[TensorLike]: The results of the simulation.
+        """
+
+        self._reset_state()
+
+        for op in circuit.operations:
+            self._apply_operation(op)
+
+        if not circuit.shots:
+            if len(circuit.measurements) == 1:
+                return self.measurement(circuit.measurements[0])
+            return tuple(self.measurement(mp) for mp in circuit.measurements)
+
+        raise NotImplementedError
+
+    def _apply_operation(self, op: qml.operation.Operator) -> None:
+        """Apply a single operator to the circuit, keeping the state always in a MPS form.
+
+        Internally it uses `quimb`'s `apply_gate` method. For operations that act on more than two wires,
+        it decomposes them first into operations that act on at most two wires.
+
+        Args:
+            op (Operator): The operation to apply.
+        """
+
+        if len(op.wires) <= 2:
+            self._circuitMPS.apply_gate(op.matrix(), *op.wires, **self._gate_opts)
+        else:
+            decom_ops = decompose_recursive(op)
+            for o in decom_ops:
+                self._circuitMPS.apply_gate(o.matrix(), *o.wires, **self._gate_opts)
+
+    def measurement(self, measurementprocess: MeasurementProcess) -> TensorLike:
+        """Measure the measurement required by the circuit over the MPS.
+
+        Args:
+            measurementprocess (MeasurementProcess): measurement to apply to the state.
+
+        Returns:
+            TensorLike: the result of the measurement.
+        """
+
+        return self._get_measurement_function(measurementprocess)(measurementprocess)
+
+    def _get_measurement_function(
+        self, measurementprocess: MeasurementProcess
+    ) -> Callable[[MeasurementProcess, TensorLike], TensorLike]:
+        """Get the appropriate method for performing a measurement.
+
+        Args:
+            measurementprocess (MeasurementProcess): measurement process to apply to the state
+
+        Returns:
+            Callable: function that returns the measurement result
+        """
+        if isinstance(measurementprocess, StateMeasurement):
+            if isinstance(measurementprocess, ExpectationMP):
+                return self.expval
+
+            if isinstance(measurementprocess, VarianceMP):
+                return self.var
+
+        raise NotImplementedError
+
+    def expval(self, measurementprocess: MeasurementProcess) -> float:
+        """Expectation value of the supplied observable contained in the MeasurementProcess.
+
+        Args:
+            measurementprocess (StateMeasurement): measurement to apply to the MPS.
+
+        Returns:
+            Expectation value of the observable.
+        """
+
+        obs = measurementprocess.obs
+
+        result = self._local_expectation(obs.matrix(), tuple(obs.wires))
```

Comment:
Will we want to turn this into an MPO eventually?

Reply:
I'm not sure. Here, we just call the [`local_expectation`](https://quimb.readthedocs.io/en/latest/autoapi/quimb/experimental/merabuilder/index.html#quimb.experimental.merabuilder.TensorNetworkGenIso.local_expectation) function in `quimb`, which returns a `float`. Can you elaborate? Thanks!

Reply:
I'm not 100% sure, but here is what I think will happen. If you have a 10-qubit Hermitian obs, you'll get a 1024 x 1024 matrix that Quimb will interpret as a 20-dimensional tensor. Even if the indices are contiguous, this will dramatically increase the intermediate tensor ranks (and computational cost). This is also true even of "natural" MPOs like tensor product operators. So we're basically straying from the MPS framework if we don't, unless Quimb magically turns the matrix into an MPO.

Reply:
For context: as discussed on Slack, `quimb` handles the matrix as an MPO. Eventually, we can come back to this in a future PR (when dealing more with benchmark and optimization)